src/fides/api/ops/service/connectors/dynamodb_connector.py

```diff
@@ -0,0 +1,175 @@
+        """Query wrapper corresponding to the input traversal_node."""
+        client = self.client()
+        try:
+            describe_table = client.describe_table(TableName=node.address.collection)
+            for key in describe_table["Table"]["KeySchema"]:
+                if key["KeyType"] == "HASH":
+                    hash_key = key["AttributeName"]
+                    break
+            for key in describe_table["Table"]["AttributeDefinitions"]:
+                if key["AttributeName"] == hash_key:
+                    attribute_definitions = [key]
+                    break
+        except ClientError as error:
+            raise ConnectorFailureException(error.response["Error"]["Message"])
+
+        return DynamoDBQueryConfig(node, attribute_definitions)
+
+    def test_connection(self) -> Optional[ConnectionTestStatus]:
+        """
+        Connects to AWS DynamoDB and lists tables to validate credentials.
+        """
+        logger.info("Starting test connection to {}", self.configuration.key)
+        client = self.client()
+        try:
+            client.list_tables()
+        except ClientError as error:
+            if error.response["Error"]["Code"] in [
+                "InvalidClientTokenId",
+                "SignatureDoesNotMatch",
+            ]:
+                raise ConnectorAuthFailureException(error.response["Error"]["Message"])
+            raise ConnectorFailureException(error.response["Error"]["Message"])
+
+        return ConnectionTestStatus.succeeded
+
+    def retrieve_data(
+        self,
+        node: TraversalNode,
+        policy: Policy,
+        privacy_request: PrivacyRequest,
+        input_data: Dict[str, List[Any]],
+    ) -> List[Row]:
+        """
+        Retrieve DynamoDB data.
+        In the case of complex objects, returns multiple rows
+        as the product of options to query against.
+        """
+        deserializer = TypeDeserializer()
+        collection_name = node.address.collection
+        client = self.client()
+        try:
+            results = []
+            query_config = self.query_config(node)
+            for attribute_definition in query_config.attribute_definitions:  # type: ignore
+                attribute_name = attribute_definition["AttributeName"]
+                for identifier in input_data[attribute_name]:
+                    selected_input_data = input_data
+                    selected_input_data[attribute_name] = [identifier]
+                    query_param = query_config.generate_query(
+                        selected_input_data, policy
+                    )
+                    items = client.query(
+                        TableName=collection_name,
+                        ExpressionAttributeValues=query_param[
+                            "ExpressionAttributeValues"
+                        ],
+                        KeyConditionExpression=query_param["KeyConditionExpression"],
```

Comment:
I'm glad you got this working!

Reply:
Yes! This one didn't have as many drawbacks from what I could tell and allowed us to account for SortKey usage as well 💙